### PR TITLE
Collect and modify push_back to emplace_back

### PR DESCRIPTION
**Pull request by boshengchen**
1. Changed from `push_back` to `emplace_back`, let avoid unnecessary memory copy/move.
2. Normalize the some codes.

### CODE REVIEW

File: src/dataman/RowReader.cpp [hunk 1]
```diff
@@ -244,14 +244,14 @@ bool RowReader::processHeader(folly::StringPiece row) {
     }
     offsets_.resize(numFields + 1, -1);
     offsets_[0] = 0;
-    blockOffsets_.push_back(std::make_pair(0, 0));
+    blockOffsets_.emplace_back(std::make_pair(0, 0));
```
Comment (dutor) [inline src/dataman/RowReader.cpp]: You should have known that `std::make_pair` is not necessary here if `emplace_back` used.

File: src/dataman/RowReader.cpp [hunk 2]
```diff
@@ -244,14 +244,14 @@ bool RowReader::processHeader(folly::StringPiece row) {
...
     blockOffsets_.reserve(numOffsets);
     for (uint32_t i = 0; i < numOffsets; i++) {
         int64_t offset = 0;
         for (int32_t j = 0; j < numBytesForOffset_; j++) {
             offset |= (uint64_t(*(it++)) << (8 * j));
         }
-        blockOffsets_.push_back(std::make_pair(offset, 0));
+        blockOffsets_.emplace_back(std::make_pair(offset, 0));
```
Comment (dutor) [inline src/dataman/RowReader.cpp]: ditto.

File: src/dataman/RowReader.cpp (hunk 1 above)
Comment (boshengchen) [reply src/dataman/RowReader.cpp]: Very careful review . I benefit greatly by this point. You are the master of c++. 
Let me address them just now.

**Comment by boshengchen** [comment]
Address dutor's comments.

**Comment by dutor** [comment]
@boshengchen I had added you into the white list to be able to trigger the CI. Please have a try.

**Comment by boshengchen** [comment]
Jenkins go.

**Comment by boshengchen** [comment]
Jenkins go

**Comment by boshengchen** [comment]
Jenkins go

**Comment by boshengchen** [comment]
Jenkins go!

**Comment by boshengchen** [comment]
Jenkins go

**Pull request merged**